backend/app/ai/utils/ai_search.py

```
@@ -48,28 +50,69 @@ def generate_keywords(query: str) -> dict:
             return json.loads(json_text.group())
         else:
+            return {"keywords": [], "keywords_en": [], "synonyms": {}}
+    except (json.JSONDecodeError, AttributeError, IndexError) as e:
+        print(f"Error parsing AI response: {e}")
+        return {"keywords": [], "keywords_en": [], "synonyms": {}}
 
-            return {"keywords": [], "keywords_en": []}
-    except Exception:
-        return {"keywords": [], "keywords_en": []}
+
+def extract_price_range_from_text(text: str) -> Union[List[float], None]:
```

Ruff (0.8.2)

Use `list` instead of `List` for type annotation

Replace with `list`

(UP006)